SingularityService/src/main/java/com/hubspot/singularity/helpers/RequestHelper.java

```diff
@@ -319,15 +310,27 @@ public void updateRequest(SingularityRequest request, Optional<SingularityReques
           entry.getValue(),
           pendingDeploy.isPresent() && pendingDeploy.get().getDeployMarker().getDeployId().equals(entry.getKey()));
       for (SingularityTaskId taskId : entry.getValue()) {
-        if (healthyTasksIdsForDeploy.contains(taskId)) {
+        if (taskManager.isKilledTask(taskId)) {
+          killedTaskIds.add(taskId);
+        } else if (healthyTasksIdsForDeploy.contains(taskId)) {
           healthyTaskIds.add(taskId);
         } else {
           notYetHealthyTaskIds.add(taskId);
         }
       }
     }
 
-    return Optional.of(new SingularityTaskIdsByStatus(healthyTaskIds, notYetHealthyTaskIds, pendingTaskIds, cleaningTaskIds));
+    List<SingularityTaskId> loadBalanced = new ArrayList<>();
+    if (requestWithState.getRequest().isLoadBalanced()) {
+      healthyTaskIds.stream()
+          .filter(taskManager::isInLoadBalancer)
+          .forEach(loadBalanced::add);
+      cleaningTaskIds.stream()
+          .filter(taskManager::isInLoadBalancer)
```

Comment:
does the task manager automatically update itself when tasks go in and out of the load balancer?

Reply:
yeah, the isInLoadBalancer here is a live call to fetch the state (cached on the leader in memory)